packages/flutter/lib/src/material/data_table.dart

```diff
@@ -1230,7 +1240,7 @@ class _SortArrowState extends State<_SortArrow> with TickerProviderStateMixin {
         transform: Matrix4.rotationZ(_orientationOffset + _orientationAnimation.value)
                              ..setTranslationRaw(0.0, _arrowIconBaselineOffset, 0.0),
         alignment: Alignment.center,
-        child: const Icon(
+        child: widget.sortIcon ?? const Icon(
```

Comment:
should this just be the default value of the constructor parameter for DataTable? That would make this a compile-time test rather than a runtime test.